hyperspy/learn/mva.py

```diff
@@ -471,202 +550,234 @@ def decomposition(self,
-            the signal. If `diff_order` is 0, the signal is not differentiated.
+            the signal. If ``diff_order`` is 0, the signal is not differentiated.
         diff_axes : None or list of ints or strings
-            If None, when `diff_order` is greater than 1 and `signal_dimension`
-            (`navigation_dimension`) when `on_loadings` is False (True) is
-            greater than 1, the differences are calculated across all
-            signal (navigation) axes. Otherwise the axes can be specified in
-            a list.
-        factors : Signal or numpy.ndarray
+            * If None and `on_loadings` is False, when `diff_order` is greater than 1
+              and `signal_dimension` is greater than 1, the differences are calculated
+              across all signal axes
+            * If None and `on_loadings` is True, when `diff_order` is greater than 1
+              and `navigation_dimension` is greater than 1, the differences are calculated
+              across all navigation axes
+            * Otherwise the axes can be specified in a list.
+        factors : :py:class:`~hyperspy.signal.BaseSignal` or numpy array
             Factors to decompose. If None, the BSS is performed on the
-            factors of a previous decomposition. If a Signal instance the
+            factors of a previous decomposition. If a Signal instance, the
             navigation dimension must be 1 and the size greater than 1.
-        comp_list : numpy.ndarray of bool
-            choose the components to use by the boolean list. It permits
-            to choose non contiguous components.
-        mask : :py:class:`~hyperspy.signal.BaseSignal` (or subclass)
+        comp_list : None or list or numpy array
+            Choose the components to apply BSS to. Unlike ``number_of_components``,
+            this argument permits non-contiguous components.
+        mask : :py:class:`~hyperspy.signal.BaseSignal` or subclass
             If not None, the signal locations marked as True are masked. The
             mask shape must be equal to the signal shape
             (navigation shape) when `on_loadings` is False (True).
-        on_loadings : bool
+        on_loadings : bool, default False
             If True, perform the BSS on the loadings of a previous
-            decomposition. If False, performs it on the factors.
-        reverse_component_criterion : str
-            One of {'factors', 'loadings'}. Use either the factor or the 
-            loading to determine if the component needs to be reversed.
-        **kwargs : extra key word arguments
+            decomposition, otherwise, perform the BSS on the factors.
+        reverse_component_criterion : {"factors", "loadings"}, default "factors"
+            Use either the factors or the loadings to determine if the
+            component needs to be reversed.
+        whiten_method : {"pca", "zca", None}, default "pca"
```

Comment:
I think it can be, and for a trivial extra two lines since we do the SVD for whitening anyway, I thought I might as well add it.

Reply:
The thing is that I think that we should favor a minimalist approach: if something makes no difference as far as we can tell, better not to implement it since it'll increase the maintenance burden. If you can provide an example where it is useful that's a different story...

Reply:
Ok sure - I'll take it out.

Reply:
No ignore that - see https://cbrnr.github.io/2018/12/17/whitening-pca-zca/ 

> A recent article by Montoya-Martínez et al. (2017) suggests that some ICA variants can be sensitive to the choice of the initial whitening procedure. Specifically, it can make a difference whether PCA or ZCA is used prior to performing Extended Infomax as implemented in EEGLAB.

Reply:
If people can now provide their own BSS algorithms (e.g. they could pass their own implementation of Extended Infomax) then perhaps having flexibility with whitening is good.

Reply:
Sounds good then.